extensions/filesystem/vault-fs/src/main/java/org/eclipse/dataspaceconnector/core/security/fs/FsPrivateKeyResolver.java

```diff
@@ -15,25 +15,37 @@
 
 package org.eclipse.dataspaceconnector.core.security.fs;
 
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
 import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.security.KeyParser;
 import org.eclipse.dataspaceconnector.spi.security.PrivateKeyResolver;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.IOException;
+import java.io.StringWriter;
 import java.security.GeneralSecurityException;
 import java.security.Key;
 import java.security.KeyStore;
 import java.security.PrivateKey;
 import java.security.interfaces.ECPrivateKey;
 import java.security.interfaces.RSAPrivateKey;
+import java.util.ArrayList;
 import java.util.Enumeration;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
 
 /**
  * Resolves an RSA or EC private key from a JKS keystore.
  */
 public class FsPrivateKeyResolver implements PrivateKeyResolver {
```

Comment:
This class resembles a lot `VaultPrivateKeyResolver`, have you considered extracting common logic in a common ancestor?

Reply:
Maybe like this: https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1644/files#diff-55f6efe408299eabee0d16795438559d844eeb69236d4b4c8ca6b6f74ff67507

Reply:
Thanks @bscholtes1A. Updated the changes accordingly to move common code at one place.